### DISCUSSION TITLE
Custom boxShadow not working when extend the theme using tailwind.config.js

### DISCUSSION THREAD

**Discussion by Chandu4221** [tailwindlabs/tailwindcss#14345] [Help]
**What version of Tailwind CSS are you using?**
`^3.4.10`

**What build tool (or framework if it abstracts the build tool) are you using?**
`"vite": "^5.4.1"`  `"postcss": "^8.4.41"`  `"tailwindcss": "^3.4.10"`

**What version of Node.js are you using?**
`node - v20.12.2`  `npm:10.5.2`

**What browser are you using?**
`Chrome - 124.0.6367.61`

**What operating system are you using?**
`Windows 10`

**Reproduction URL**

`package.json` dependencies and dev dependencies
```json
{
 "dependencies": {
    "@radix-ui/react-slot": "^1.1.0",
    "class-variance-authority": "^0.7.0",
    "clsx": "^2.1.1",
    "lucide-react": "^0.436.0",
    "react": "^18.3.1",
    "react-dom": "^18.3.1",
    "tailwind-merge": "^2.5.2",
    "tailwindcss-animate": "^1.0.7"
  },
  "devDependencies": {
    "@eslint/js": "^9.9.0",
    "@types/node": "^22.5.0",
    "@types/react": "^18.3.3",
    "@types/react-dom": "^18.3.0",
    "@vitejs/plugin-react": "^4.3.1",
    "autoprefixer": "^10.4.20",
    "eslint": "^9.9.0",
    "eslint-plugin-react-hooks": "^5.1.0-rc.0",
    "eslint-plugin-react-refresh": "^0.4.9",
    "globals": "^15.9.0",
    "postcss": "^8.4.41",
    "tailwindcss": "^3.4.10",
    "typescript": "^5.5.3",
    "typescript-eslint": "^8.0.1",
    "vite": "^5.4.1"
  }
  }
```
[Tailwind Playground working here](https://play.tailwindcss.com/SYIVUe6jCr)

**Describe your issue**
I am using `create vite@latest` with `react-ts` and installed `shadcn-ui`

I have tried extending the `boxShadow` class in the `tailwind.config.js` (refer below image)

![image](https://github.com/user-attachments/assets/942844d3-83b9-424f-bcdf-036fa7ff2203)

I am also getting suggestion in vscode when i hover on the class pop up with the css rule is shown (refer below image)

![image](https://github.com/user-attachments/assets/66608083-cc8e-426a-9e05-6212bf0fcc51)

but when i inspect the element on webpage my custom shadow class is missing (refer below image)

![image](https://github.com/user-attachments/assets/a062be58-d190-43df-ace3-c3247a31d4c3)

---

When i modify the `key` in `tailwind.config.js` (refer below image)

![image](https://github.com/user-attachments/assets/f8b00637-eda1-414e-9033-438d1154be00)

and apply the class to my component 
![image](https://github.com/user-attachments/assets/690e8d69-fcc4-489b-bbe0-c5959fea70f4)

and inspect the component in web page (refer below image)
it is working as expected.
![image](https://github.com/user-attachments/assets/b4e5823b-08f6-4dbf-b758-ed42965c8e5c)

**Comment by wongjn** [comment]
It could be that you'd need to configure `tailwind-merge` to understand your custom class name in the `Card` component. This is so that it can merge the class name list properly.

**Reply by Chandu4221** [reply]
@wongjn Why is `4xl` working without configuring `tailwind-merge`, but `brutal` is not?. Please explain.

**Reply by wongjn** [reply]
Perhaps `tailwind-merge` has some built-in heuristics that interprets `4xl` differently to `brutal`.

**Comment by LinThitHtwe** [comment]
How about adding important in there ?

```
!shadow-brutal
```

**Comment by david-ouatedem** [comment]
> **What version of Tailwind CSS are you using?** `^3.4.10`
> 
> **What build tool (or framework if it abstracts the build tool) are you using?** `"vite": "^5.4.1"` `"postcss": "^8.4.41"` `"tailwindcss": "^3.4.10"`
> 
> **What version of Node.js are you using?** `node - v20.12.2` `npm:10.5.2`
> 
> **What browser are you using?** `Chrome - 124.0.6367.61`
> 
> **What operating system are you using?** `Windows 10`
> 
> **Reproduction URL**
> 
> `package.json` dependencies and dev dependencies
> 
> ```json
> {
>  "dependencies": {
>     "@radix-ui/react-slot": "^1.1.0",
>     "class-variance-authority": "^0.7.0",
>     "clsx": "^2.1.1",
>     "lucide-react": "^0.436.0",
>     "react": "^18.3.1",
>     "react-dom": "^18.3.1",
>     "tailwind-merge": "^2.5.2",
>     "tailwindcss-animate": "^1.0.7"
>   },
>   "devDependencies": {
>     "@eslint/js": "^9.9.0",
>     "@types/node": "^22.5.0",
>     "@types/react": "^18.3.3",
>     "@types/react-dom": "^18.3.0",
>     "@vitejs/plugin-react": "^4.3.1",
>     "autoprefixer": "^10.4.20",
>     "eslint": "^9.9.0",
>     "eslint-plugin-react-hooks": "^5.1.0-rc.0",
>     "eslint-plugin-react-refresh": "^0.4.9",
>     "globals": "^15.9.0",
>     "postcss": "^8.4.41",
>     "tailwindcss": "^3.4.10",
>     "typescript": "^5.5.3",
>     "typescript-eslint": "^8.0.1",
>     "vite": "^5.4.1"
>   }
>   }
> ```
> 
> [Tailwind Playground working here](https://play.tailwindcss.com/SYIVUe6jCr)
> 
> **Describe your issue** I am using `create vite@latest` with `react-ts` and installed `shadcn-ui`
> 
> I have tried extending the `boxShadow` class in the `tailwind.config.js` (refer below image)
> 
> ![image](https://private-user-images.githubusercontent.com/17278323/364434070-942844d3-83b9-424f-bcdf-036fa7ff2203.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU5ODIyODcsIm5iZiI6MTcyNTk4MTk4NywicGF0aCI6Ii8xNzI3ODMyMy8zNjQ0MzQwNzAtOTQyODQ0ZDMtODNiOS00MjRmLWJjZGYtMDM2ZmE3ZmYyMjAzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTEwVDE1MjYyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVjMzc0MTUxMjA1M2QwZTJkMzgzNDdiZjY0NjBkNDZkZWI4MWQxYjQ3YzE4NmIwMmNhZTBmZWQ1YmVjNmZmY2UmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.262aFtBkRvdwefpC0CuG3LoaraMrVbE90WjJdS_y_YY)
> 
> I am also getting suggestion in vscode when i hover on the class pop up with the css rule is shown (refer below image)
> 
> ![image](https://private-user-images.githubusercontent.com/17278323/364434985-66608083-cc8e-426a-9e05-6212bf0fcc51.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU5ODIyODcsIm5iZiI6MTcyNTk4MTk4NywicGF0aCI6Ii8xNzI3ODMyMy8zNjQ0MzQ5ODUtNjY2MDgwODMtY2M4ZS00MjZhLTllMDUtNjIxMmJmMGZjYzUxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTEwVDE1MjYyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNjZWM5ZjAyM2Q2MTIwZTljNmE0NTM5ZjdiNmQ0MDM1NmNjZjM2OTI1YmJlYjcxMTVmMDAxZTVmNWI5NDY3NTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.RPOq0v0Lp4tZUeOLUQvQc_uYMvv3AtnEYX7pgFne1gk)
> 
> but when i inspect the element on webpage my custom shadow class is missing (refer below image)
> 
> ![image](https://private-user-images.githubusercontent.com/17278323/364435500-a062be58-d190-43df-ace3-c3247a31d4c3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU5ODIyODcsIm5iZiI6MTcyNTk4MTk4NywicGF0aCI6Ii8xNzI3ODMyMy8zNjQ0MzU1MDAtYTA2MmJlNTgtZDE5MC00M2RmLWFjZTMtYzMyNDdhMzFkNGMzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTEwVDE1MjYyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYwOTEwN2I1OTkyMTk2YmE0NGE2Mzg1OTA2YzUwODhiZTNkNTA5M2Q5MDM4YThiMmMxMWQ0MWEwOWFhM2M3YmYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.qhMCxMkp_VtWDjXA3zbc7-2mPTnNDOJqFalHkSab6J8)
> 
> When i modify the `key` in `tailwind.config.js` (refer below image)
> 
> ![image](https://private-user-images.githubusercontent.com/17278323/364441575-f8b00637-eda1-414e-9033-438d1154be00.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU5ODIyODcsIm5iZiI6MTcyNTk4MTk4NywicGF0aCI6Ii8xNzI3ODMyMy8zNjQ0NDE1NzUtZjhiMDA2MzctZWRhMS00MTRlLTkwMzMtNDM4ZDExNTRiZTAwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTEwVDE1MjYyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZmYmUzMTc3YTUyMGQ0YWYzODcxMDcyNGVkZjQwY2RlY2FkMDBmNTMyMTNjNzFiY2RhOWU0MTVjZTVkZGNhYjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.kV-BH61Za8l9ZgN02MA_NoXtfzA2ggAUtJ_NcrOnUQI)
> 
> and apply the class to my component ![image](https://private-user-images.githubusercontent.com/17278323/364442147-690e8d69-fcc4-489b-bbe0-c5959fea70f4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU5ODIyODcsIm5iZiI6MTcyNTk4MTk4NywicGF0aCI6Ii8xNzI3ODMyMy8zNjQ0NDIxNDctNjkwZThkNjktZmNjNC00ODliLWJiZTAtYzU5NTlmZWE3MGY0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTEwVDE1MjYyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUxZTFlZGE5YmQwNjZjYmEzZGZjYjZhOWMzNzJiNDYyZTRhOTk3MzM5NDE3Mjg3MTdmNjM4OTBkNDU5M2IxODgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.gFCTJnwFX33JPXnNUNzlqt6xmVQyccY9Z342fH8jjEw)
> 
> and inspect the component in web page (refer below image) it is working as expected. ![image](https://private-user-images.githubusercontent.com/17278323/364442809-b4e5823b-08f6-4dbf-b758-ed42965c8e5c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU5ODIyODcsIm5iZiI6MTcyNTk4MTk4NywicGF0aCI6Ii8xNzI3ODMyMy8zNjQ0NDI4MDktYjRlNTgyM2ItMDhmNi00ZGJmLWI3NTgtZWQ0Mjk2NWM4ZTVjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTEwVDE1MjYyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI3NGI3MzQ4ZDhjMmMzYjNmMTMxNjA5MzE0NmMxZDdlOGQwODIyYTViNmVlOGJmZmYzOTNkZWNlYzk5Zjc5MjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Lys8BR7XwNIWgqQ3R1zFRqLBoSh1WLrlRuI__3dcrIw)

Your class `shadow-blue-300` is overwriting the class `shadow-brutal` remove it

**Reply by Chandu4221** [reply]
@david-ouatedem `shadow-blue-300` this just adds shadow color. Please refer [BoxShadow Docs](https://tailwindcss.com/docs/box-shadow-color), also check the working example attached

**Reply by david-ouatedem** [reply]
I say it overwrites because when I tried your example the style for `shadow-brutal` did not apply when I had `shadow-blue-300`. I don't know how tailwindcss handles the merging of custom styles from `tailwind.config.ts` and tailwind classes but that is how it worked for me
NB: we have the same versions of tailwind-related dependencies

**Reply by Chandu4221** [reply]
@david-ouatedem have you tried `shadow-4xl` example ? it works as expected. My class `shadow-brutal` just adds a shadow to the component, while `shadow-blue-300` sets the shadow color.

The Tailwind CSS plugin usually warns when trying to modify the same property with two classes (CSS conflict, i have used `shadow-blue-300` and `shadow-red-300` trying to modify same css property), but that’s not happening in my `shadow-brutal` case. My `shadow-brutal` class adds the shadow, and `shadow-blue-300` sets the shadow color.
(refer below image)
![image](https://github.com/user-attachments/assets/ae979ef5-6004-408c-932a-e8ffceacb516)
